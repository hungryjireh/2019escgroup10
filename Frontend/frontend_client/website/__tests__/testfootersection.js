import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import FooterSection from "../components/Footer/FooterSection";
import {SectionHeader, SectionList, ListItem, ItemLink} from "../components/Footer/FooterSection";

afterEach(cleanup);

it("SectionHeader renders", () => {
    const { asFragment } = render(<SectionHeader />);
    expect(asFragment()).toMatchSnapshot();
});

it("SectionList renders", () => {
    const { asFragment } = render(<SectionList />);
    expect(asFragment()).toMatchSnapshot();
});

it("ListItem renders", () => {
    const { asFragment } = render(<ListItem />);
    expect(asFragment()).toMatchSnapshot();
});

it("ItemLink renders", () => {
    const { asFragment } = render(<ItemLink />);
    expect(asFragment()).toMatchSnapshot();
});

test("ItemLink renders with alphanum inputs", () => {
    const { getByText } = render(<ItemLink>Hello55!</ItemLink>);
    expect(getByText("Hello55!")).toBeTruthy;
});

test("ItemLink renders with image inputs", () => {
    const renderComponent = () => render(<ItemLink><img></img></ItemLink>);
    expect(renderComponent).toBeTruthy;
});

test("SectionHeader renders with alphanum inputs", () => {
    const { getByText } = render(<SectionHeader>Hello55!</SectionHeader>);
    expect(getByText("Hello55!")).toBeTruthy;
});

test("SectionHeader renders with image inputs", () => {
    const renderComponent = () => render(<SectionHeader><img></img></SectionHeader>);
    expect(renderComponent).toBeTruthy;
});

test("SectionList renders with alphanum inputs", () => {
    const { getByText } = render(<SectionList>Hello55!</SectionList>);
    expect(getByText("Hello55!")).toBeTruthy;
});

test("SectionList renders with image inputs", () => {
    const renderComponent = () => render(<SectionList><img></img></SectionList>);
    expect(renderComponent).toBeTruthy;
});

test("ListItem renders with alphanum inputs", () => {
    const { getByText } = render(<ListItem>Hello55!</ListItem>);
    expect(getByText("Hello55!")).toBeTruthy;
});

test("ListItem renders with image inputs", () => {
    const renderComponent = () => render(<ListItem><img></img></ListItem>);
    expect(renderComponent).toBeTruthy;
});