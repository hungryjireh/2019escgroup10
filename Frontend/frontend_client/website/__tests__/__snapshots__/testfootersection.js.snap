// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemLink renders 1`] = `
<DocumentFragment>
  <a
    class="sc-ifAKCX heBHq"
  />
</DocumentFragment>
`;

exports[`ListItem renders 1`] = `
<DocumentFragment>
  <li
    class="sc-bxivhb giIgxL"
  />
</DocumentFragment>
`;

exports[`SectionHeader renders 1`] = `
<DocumentFragment>
  <h3
    class="sc-bwzfXH kGzkEe"
  />
</DocumentFragment>
`;

exports[`SectionList renders 1`] = `
<DocumentFragment>
  <ul
    class="sc-htpNat dePyzG"
  />
</DocumentFragment>
`;
